core/views/api.py

```diff
@@ -0,0 +1,157 @@
+import datetime
+
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.paginator import InvalidPage, Paginator
+from django.http import JsonResponse
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.decorators import api_view
+
+from core import models, rest_serializers
+
+
+@api_view(['GET'])
+def awardee(request, org_code):
+    """
+    api/chronam/awardee/<org_code>.json
+    Retrieve an awardee and its batches
+    """
+    try:
+        awardee = models.Awardee.objects.get(org_code=org_code)
+    except ObjectDoesNotExist:
+        return JsonResponse({'detail': 'Awardee does not exist'}, status=status.HTTP_404_NOT_FOUND)
+    serializer = rest_serializers.AwardeeSerializer(awardee, context={'request': request})
+    return JsonResponse(serializer.data, safe=False)
+
+
+@api_view(['GET'])
+def awardee_list(request):
+    """
+    api/chronam/awardees.json
+    List all awardees
+    """
+    awardees = models.Awardee.objects.all()
+    serializer = rest_serializers.AwardeeListSerializer(awardees, many=True, context={'request': request})
+    return JsonResponse({'awardees': serializer.data}, safe=False)
+
+
+@api_view(['GET'])
+def batch(request, batch_name):
+    """
+    api/chronam/batches/<batch_name>.json
+    Retrieve a batch and its issues
+    """
+    try:
+        batch = models.Batch.objects.get(name=batch_name)
+    except ObjectDoesNotExist:
+        return JsonResponse({'detail': 'Batch does not exist'}, status=status.HTTP_404_NOT_FOUND)
+    serializer = rest_serializers.BatchSerializer(batch, context={'request': request})
+    return JsonResponse(serializer.data, safe=False)
+
+
+@api_view(['GET'])
+def batch_list(request, page_number=1):
+    """
+    api/chronam/batches.json
+    List all batches
+    """
+    try:
+        batches = models.Batch.objects.all().order_by('name')
```

Comment:
Need to fix this so it sorts consistently with the HTML view (I think it's by create date descending, but double-check that)

Reply:
Wonder if there's a way to share code so this is a non-issue. Having the exact same top-level "find, sort, paginate" logic seems bad.... Probably a future ticket though.

Reply:
Fixed this, but didn't look into centralizing the ordering somewhere. If we do that, might be wiser as part of a broader effort than just this one pair of data representations.